src/verifier.js

```diff
@@ -22,81 +22,94 @@ function Verifier(providerBaseUrl, pactUrls, providerStatesUrl, providerStatesSe
+				'pactUrls': '--pact-urls',
+				'providerStatesUrl': '--provider-states-url',
+				'providerStatesSetupUrl': '--provider-states-setup-url',
+				'pactBrokerUsername': '--broker-username',
+				'pactBrokerPassword': '--broker-password',
+				'publishVerificationResult': '--publish-verification-results',
+				'providerVersion': '--provider-app-version'
+			});
+
+		var cmd = [verifierPath.file].concat(args).join(' ');
+
+		if (isWindows) {
+			file = 'cmd.exe';
+			args = ['/s', '/c', cmd];
+			opts.windowsVerbatimArguments = true;
+		} else {
+			cmd = './' + cmd;
+			file = '/bin/sh';
+			args = ['-c', cmd];
+		}
+
+		self._instance = cp.spawn(file, args, opts);
+
+		self._instance.stdout.setEncoding('utf8');
+		self._instance.stdout.on('data', outputHandler);
+		self._instance.stderr.setEncoding('utf8');
+		self._instance.stderr.on('data', outputHandler);
+		self._instance.on('error', logger.error.bind(logger));
+
+		self._instance.once('close', function (code) {
+			code == 0 ? deferred.resolve(output) : deferred.reject(new Error(output));
 		});
 
-	var cmd = [verifierPath.file].concat(args).join(' ');
-
-	if (isWindows) {
-		file = 'cmd.exe';
-		args = ['/s', '/c', cmd];
-		opts.windowsVerbatimArguments = true;
-	} else {
-		cmd = './' + cmd;
-		file = '/bin/sh';
-		args = ['-c', cmd];
-	}
-
-	this._instance = cp.spawn(file, args, opts);
-
-	this._instance.stdout.setEncoding('utf8');
-	this._instance.stdout.on('data', outputHandler);
-	this._instance.stderr.setEncoding('utf8');
-	this._instance.stderr.on('data', outputHandler);
-	this._instance.on('error', logger.error.bind(logger));
-
-	this._instance.once('close', function (code) {
-		code == 0 ? deferred.resolve(output) : deferred.reject(new Error(output));
+		logger.info('Created Pact Verifier process with PID: ' + self._instance.pid);
+		return deferred.promise.timeout(self._options.timeout, "Timeout waiting for verification process to complete (PID: " + self._instance.pid + ")")
+			.tap(function (data) {
+				logger.info('Pact Verification succeeded.');
+			});
 	});
-
-	logger.info('Created Pact Verifier process with PID: ' + this._instance.pid);
-	return deferred.promise.timeout(this._options.timeout, "Timeout waiting for verification process to complete (PID: " + this._instance.pid + ")")
-		.tap(function (data) {
-			logger.info('Pact Verification succeeded.');
-		});
 };
 
 // Creates a new instance of the pact server with the specified option
 module.exports = function (options) {
 	options = options || {};
 	options.providerBaseUrl = options.providerBaseUrl || '';
+	options.pactBrokerUrl = options.pactBrokerUrl || '';
+	options.tags = options.tags || [];
+	options.provider = options.provider || '';
 	options.pactUrls = options.pactUrls || [];
 	options.providerStatesUrl = options.providerStatesUrl || '';
 	options.providerStatesSetupUrl = options.providerStatesSetupUrl || '';
 	options.timeout = options.timeout || 30000;
+	var retrievePactsPromise;
```

Comment:
This promise, I'm not sure if it belongs here.  I think it would be better in the constructor instead of where we do all the options assertion.